src/basic/capability-util.c

```diff
@@ -364,13 +364,14 @@ bool ambient_capabilities_supported(void) {
         return cache;
 }
 
```

Comment:
hmm, i'd prefer if we had helper call: capability_quintet_mangle() or so, that takes a CapabilityQuintet structure and masks all bits out that are not in the current bounding set, and returns > 0 if it turned any off and == 0 if not. Then we could just call this early in nspawn, and if it returns > 0 we warn loudly (or maybe as a tweak only warn loudly if any caps where configured at all).

```c
int capability_quitntet_mangle(CapabilityQuintet *q) {
        unsigned long i;
        uint64_t combined, drop = 0;
        bool changed = false;

        assert(q);

        /* Mangles the specified caps quintet taking the current bounding set into account: drops all caps from all five sets if our bounding set doesn't allow them. */

        combined = q->effective | q->bounding | q->inheritable | q->permitted | q->ambient;

        for (i = 0; i <= cap_last_cap(); i++) {
                unsigned long bit = UINT64_C(1) << i;
                if (!FLAGS_SET(combined, bit)
                        continue;

                if (prctl(PR_CAPBSET_READ, i) > 0)
                        continue;

                drop |= bit;
        }

        q->effective &= ~drop;
        q->bounding &= ~drop;
        q->inheritable &= ~drop;
        q->permitted &= ~drop;
        q->ambient &= ~drop;

        return drop != 0; /* Let the caller know we changed something */
}
```